test_Budget_pytest.py

```diff
@@ -1,96 +1,167 @@
+import pytest
 
-def test_export_python():
-    runner = CliRunner()
-    mock_output = [
+MOCK_OUTPUT_2_ITEMS = [
         Expense(id=1, amount=34.0, description="Hit Box"),
         Expense(id=2, amount=453.0, description="Crazy curry"),
     ]
-    with patch("main.read_db_or_init", return_value=mock_output):
+
+LIST_OF_ITEMS = [(987,"Chicken"),(456,"Bananas"),(789,"Pizza")]
+
+
+@pytest.fixture
+def runner():
+   return CliRunner() 
+
+@pytest.fixture
+def use_temp_csv():
+    filename_csv = "temp.csv"
+    with open(filename_csv, "w") as file:
+        file.write('amount,description\n150,Test CSV 1\n300,Test CSV 2\n')  
+    yield filename_csv
+    os.remove(filename_csv)
+
+
+@pytest.fixture
+def use_temp_db():
+    filename_db = "test_db.json"
+    with open(filename_db, "w") as file:
+        file.write("[]")  
+    yield filename_db
+    os.remove(filename_db)
+
+# @pytest.fixture
+# def teardown_temp_db():
+#     yield
+#     os.remove("test_db.json")
+
+
+
+def test_export_python(runner: CliRunner):
+    """ Test for returning correct repr string"""
+    with patch("main.read_db_or_init", return_value=MOCK_OUTPUT_2_ITEMS):
         result = runner.invoke(export_python)
         printed_output = result.output.strip()
-        assert printed_output == repr(mock_output)
+        assert printed_output == repr(MOCK_OUTPUT_2_ITEMS)
         assert result.exit_code == 0
 
 
-def test_import_csv():
-    runner = CliRunner()
-    with tempfile.NamedTemporaryFile(delete=False, mode="w", suffix=".csv") as temp_csv:
-        temp_csv.write("amount,description\n34,Hit Box\n453,Crazy curry\n")
+def test_import_csv(runner: CliRunner):
+    """ Test to check if csv file is imported correctly and
+     if there is appropriate interaction with database"""
+    with runner.isolated_filesystem():
+        with open('test.csv', 'w') as file:
+            file.write('amount,description\n50,Test 1\n75,Test 2\n')
 
-    mock_output = [
-        Expense(id=1, amount=34.0, description="Hit Box"),
-        Expense(id=2, amount=453.0, description="Crazy curry"),
-    ]
-
-    with patch("main.add_csv_to_db", return_value=mock_output), patch(
-        "main.save_db"
-    ) as mock_save_db:
-        result = runner.invoke(import_csv, [temp_csv.name])
-        mock_save_db.assert_called_once_with(mock_output)
-        printed_output = result.output.strip()
-        assert printed_output == "Pomyślnie zaimportowano"
-    os.remove(temp_csv.name)
+        result = runner.invoke(import_csv, ['test.csv'])
+        assert result.exit_code == 0  
+        assert 'Pomyślnie zaimportowano' in result.output
 
 
-def test_report():
-    runner = CliRunner()
-    mock_output = [
-        Expense(id=1, amount=34.0, description="Hit Box"),
-        Expense(id=2, amount=453.0, description="Crazy curry"),
-    ]
-    with patch("main.read_db_or_init", return_value=mock_output):
+def test_report(runner: CliRunner):
+    """ Test to check if click module runs correctly
```

Comment:
Tutaj lepiej rozwinąć nazwę funcji samego testu. Jeżeli wybierzesz komentarze, to warto umieszczać je zawsze przed testem albo za testem :). Jeszcze lepiej docstringi umieszczać w funkcjach pomocniczych, ale nie w samych testach.